flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -1014,7 +1014,7 @@ public void resumeConsumption(InputChannelInfo channelInfo) throws IOException {
     @Override
     public void acknowledgeAllRecordsProcessed(InputChannelInfo channelInfo) throws IOException {
         checkState(!isFinished(), "InputGate already finished.");
-        if (!enabledTieredStorage()) {
+        if (consumedPartitionType.isPipelinedOrPipelinedBoundedResultPartition()) {
```

Comment:
Why the `SingleInputGate` only needs to acknowledge that all records are processed when the partition type is pipelined?